nixos/modules/programs/fish.nix

```diff
@@ -169,6 +169,71 @@ in
       end
     '';
 
+    programs.fish.interactiveShellInit = ''
+      # add completions generated by NixOS to $fish_complete_path
+      begin
+        # joins with null byte to acommodate all characters in paths, then respectively gets all paths before / after the first one including "generated_completions",
+        # splits by null byte, and then removes all empty lines produced by using 'string'
+        set -l prev (string join0 $fish_complete_path | string match --regex "^.*?(?=\x00[^\x00]*generated_completions.*)" | string split0 | string match -er ".")
+        set -l post (string join0 $fish_complete_path | string match --regex "[^\x00]*generated_completions.*" | string split0 | string match -er ".")
+        set fish_complete_path $prev "/etc/fish/generated_completions" $post
+      end
+    '';
+
+    environment.etc."fish/generated_completions".source =
+      let
+        patchedGenerator = pkgs.stdenv.mkDerivation {
+          name = "fish_patched-completion-generator";
+          srcs = [
+            "${pkgs.fish}/share/fish/tools/create_manpage_completions.py"
+            "${pkgs.fish}/share/fish/tools/deroff.py"
+          ];
+          unpackCmd = "cp $curSrc $(basename $curSrc)";
+          sourceRoot = ".";
+          patches = [ (pkgs.writeText "fish_completion-generator_patch" ''
+            --- a/create_manpage_completions.py
+            +++ b/create_manpage_completions.py
+            @@ -776,8 +776,6 @@ def parse_manpage_at_path(manpage_path, output_directory):
+
+                         built_command_output.insert(0, "# " + CMDNAME)
+
+            -            # Output the magic word Autogenerated so we can tell if we can overwrite this
+            -            built_command_output.insert(1, "# Autogenerated from man page " + manpage_path)
+                         # built_command_output.insert(2, "# using " + parser.__class__.__name__) # XXX MISATTRIBUTES THE CULPABILE PARSER! Was really using Type2 but reporting TypeDeroffManParser
+
+                         for line in built_command_output:
+          '') ];
+          dontBuild = true;
+          installPhase = ''
+            mkdir -p $out
+            cp * $out/
+          '';
+        };
+        generateCompletions = package: pkgs.runCommand
+          "${package.name}_fish-completions"
```

Comment:
I preferred the dash. But I guess that's up to you :)

Reply:
I thought it's better to separate it clearly from the name of the package which usually already includes dashes.